File: DesignPatternsExercises/java/InterpreterDesignPattern/Solution/SubtractionExpression.java
public class SubtractionExpression extends Expression {

    public SubtractionExpression(Constant firstConstant, Constant secondConstant) {
        // __________(16)__________;
        super(firstConstant, secondConstant);
    }

    @Override
    public int interpret() {
        // return firstConstant.interpret() __________(17)__________ secondConstant.interpret();
        return firstConstant.interpret() - secondConstant.interpret();
    }
}

File: DesignPatternsExercises/ruby/interpreter_design_pattern/solution/subtraction_expression.rb
require_relative 'expression'

class SubtractionExpression < Expression
  def initialize(first_constant, second_constant)
    # __________(13)__________(first_constant, second_constant)
    super(first_constant, second_constant)
  end

  def interpret
    # @first_constant.interpret __________(14)__________ @second_constant.interpret
    @first_constant.interpret - @second_constant.interpret
  end
end
